src/lib/utils.spec.js

```diff
@@ -178,3 +189,53 @@ describe('getUserNameFromUrl', () => {
     expect(getUserNameFromUrl()).toBeNull()
   })
 })
+
+describe('generateReturnUrlToNotesIndex', () => {
+  function createFetchUrl() {
+    return jest.fn().mockImplementation(({ _id }) => ({
+      data: {
+        type: 'io.cozy.notes.url',
+        id: _id,
+        note_id: _id,
+        subdomain: 'flat',
+        protocol: 'https',
+        instance: 'alice.cozy.example',
+        public_name: 'Bob'
+      }
+    }))
+  }
+
+  function createClient() {
+    return {
+      getStackClient: () => ({
+        collection: () => ({ fetchURL: createFetchUrl() })
+      })
+    }
+  }
+
+  function createNote(id = '12345') {
+    return { id, type: 'io.cozy.files' }
+  }
+
+  afterEach(() => {
+    restoreLocation()
+  })
```

Comment:
Nit: I find it strange that we restore location after each test, but only alter it in some of them. Either we want the same behavior all the time and it would make sense to call `setLocation` in a `beforeEach`, or it's on-demand and we can call `restoreLocation` in the tests that need it?

Reply:
Restoring location is harmless, even if you don't setLocation before. A afterEach allows to not forget anything.